library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/mixin/client/ArmorFeatureRendererMixin.java

```diff
@@ -0,0 +1,106 @@
+
+import net.minecraft.client.render.VertexConsumerProvider;
+import net.minecraft.client.render.entity.feature.ArmorFeatureRenderer;
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ArmorItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.rendering.entity.api.client.ArmorTextureUtils;
+import org.quiltmc.qsl.rendering.entity.impl.client.ArmorRenderingRegistryImpl;
+
+@SuppressWarnings("rawtypes")
+@Mixin(ArmorFeatureRenderer.class)
+public abstract class ArmorFeatureRendererMixin {
+	@Shadow
+	@Final
+	private static Map<String, Identifier> ARMOR_TEXTURE_CACHE;
+
+	@Unique
+	private LivingEntity quilt$capturedEntity;
+	@Unique
+	private EquipmentSlot quilt$capturedSlot;
+
+	@Inject(method = "render(Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/client/render/VertexConsumerProvider;ILnet/minecraft/entity/LivingEntity;FFFFFF)V",
+			at = @At("HEAD"))
+	private void quilt$captureEntity(MatrixStack matrixStack, VertexConsumerProvider vertexConsumerProvider, int i, LivingEntity livingEntity, float f, float g, float h, float j, float k, float l, CallbackInfo ci) {
+		this.quilt$capturedEntity = livingEntity;
+	}
+
+	@Inject(method = "renderArmor", at = @At("HEAD"))
+	private void quilt$captureSlot(MatrixStack matrices, VertexConsumerProvider vertexConsumers, LivingEntity livingEntity, EquipmentSlot slot, int i, BipedEntityModel bipedEntityModel, CallbackInfo ci) {
+		this.quilt$capturedSlot = slot;
+	}
+
+	@Inject(method = "render(Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/client/render/VertexConsumerProvider;ILnet/minecraft/entity/LivingEntity;FFFFFF)V",
+			at = @At("RETURN"))
+	private void quilt$uncapture(MatrixStack matrixStack, VertexConsumerProvider vertexConsumerProvider, int i, LivingEntity livingEntity, float f, float g, float h, float j, float k, float l, CallbackInfo ci) {
+		this.quilt$capturedEntity = null;
+		this.quilt$capturedSlot = null;
+	}
+
+	@Inject(method = "getArmor", at = @At("RETURN"), cancellable = true)
+	private void quilt$getArmorModel(EquipmentSlot slot, CallbackInfoReturnable<BipedEntityModel<LivingEntity>> cir) {
+		ItemStack stack = this.quilt$capturedEntity.getEquippedStack(slot);
+
+		BipedEntityModel<LivingEntity> model = cir.getReturnValue();
+		model = ArmorRenderingRegistryImpl.getArmorModel(model, this.quilt$capturedEntity, stack, this.quilt$capturedSlot);
+		cir.setReturnValue(model);
+	}
+
+	@Inject(method = "getArmorTexture",
+			at = @At(value = "INVOKE", target = "Ljava/util/Map;computeIfAbsent(Ljava/lang/Object;Ljava/util/function/Function;)Ljava/lang/Object;"),
+			cancellable = true)
+	private void quilt$getArmorTexture(ArmorItem armorItem, boolean useSecondTexture, /* @Nullable */ String suffix, CallbackInfoReturnable<Identifier> cir) {
```

Comment:
Why is this `@Nullable` commented?

Reply:
Not sure! I think this method's signature is copied directly from Fabric.

Reply:
Ah that explains it then! Surely some javax stuff, uncommenting it would be nice (keepign it would be better)

Reply:
oops